radix-substate-store-queries/src/typed_substate_layout.rs

```diff
@@ -428,7 +428,7 @@ pub enum TypedMainModuleSubstateValue {
     NonFungibleVault(NonFungibleVaultTypedSubstateValue),
     ConsensusManager(ConsensusManagerTypedSubstateValue),
     Validator(ValidatorTypedSubstateValue),
-    AccessController(AccessControllerTypedSubstateValue),
+    AccessController(AccessControllerV2TypedSubstateValue),
```

Comment:
We need both `AccessControllerV1` and `AccessControllerV2`?

Reply:
I do not think that v1 is needed here since both v1 and v2 substates can be represented as `AccessControllerV2TypedSubstateValue`.

Reply:
Can we decode the V1 sbor payload into `AccessControllerV2TypedSubstateValue`?

Reply:
Yup should be possible since the main change is that we added a new version (enum variant) so we should be able to decode v1 as v2, I've just added a test for that: `access_controller_state_v1_can_be_decoded_as_v2`.

Also, this is implicitly tested in `test_all_scenario_commit_receipts_should_have_substate_changes_which_can_be_typed` since our scenarios use AccessControllerV1 and their state can be typed.